### PR TITLE
feat: bundle project

### PR DESCRIPTION
**Pull request by vuson1709**
### Issues

[As a customer I want to embeded the code to my website, So I can use the bot (4)](https://app.asana.com/0/1207458980633630/1207359608169810/f)

### Updated

- Install cssInjectedByJsPlugin package to combine CSS into single JS file
- Instead of create #root, using 'div' as a container for project
- Using `rem` unit for all properties in CSS
- Apply `transparent` color on body so that client's website can display proper `background-color`

### Result

Note: All links below are used for demonstration only
- Assuming this website is used for marketing Chatbot project: https://drycleaning-aichatbot.netlify.app/
- This path contains bundle file from Chatbot after running `pnpm run build`: https://drycleaning-aichatbot.netlify.app/widget/chat-widget.iife.js
- Client inject this script to their website to use Chatbot: https://son-moviedatabase.netlify.app/
```html
<script src="https://drycleaning-aichatbot.netlify.app/widget/chat-widget.iife.js"></script>
``` 

image

### CODE REVIEW

**Comment by vuson1709** [comment]
Thank you for good questions, most of it I've mentioned in the comment section of my [task](https://app.asana.com/0/1207458980633630/1207359608169810/f)

From my understand, when we start selling the product to customer, we need 4 components (2 Frontend & 2 Backend):

> **Components and Their Roles:**
> 1. Front-End Widget (A1):
>     * Displays the chat widget.
>     * Receives user input and displays output messages.
>     * Contacts B2 to verify botId.
> 2. Back-End Service for Chat Responses (B1):
>     * Connects to LLM (Language Model) to generate responses based on user input.
> 3. Marketing Site and Hosting for JS Bundle (A2):
>     * Hosts your marketing content.
>     * Hosts the JavaScript bundle for the chat widget.
> 4. Back-End Service for Bot ID Management (B2):
>     * Generates and assigns unique botId for clients upon registration.
>     * Verifies the validity of botId when contacted by A1.


> 1. Have you prepared a list of what to do for the script tag to be hosted live? So whenever we have an update in the production's UI, it will be reflected on our client's website.

- We create a landing page to marketing our product (A2), this also a host where we save our bundle file. Whenever we have an update to our product, we just send the `.js` bundle file to A2. The client will receive the update immediately without doing anything.

> 2. Another important point is that your script tag is missing some important attributes such as `botId` and `sessionId`. We need those in order to differentiate the clients.

- Based on my research above, after client registers to our product (A2), the backend (B2) will provides the client the script and it also responses for generate unique `botId`. When they initialize the Chatbot (A1), A1 will read the `botId` attribute from html and will contact backend (B2) to confirm that this `botId` is valid, if so, A1 will connect to B1 to work with LLM.

**Comment by higherstates** [comment]
@vuson1709 let's talk about this before merging!

> Based on my research above, after client registers to our product (A2), the backend (B2) will provides the client the script and it also responses for generate unique botId. When client initializes the Chatbot (A1), A1 will read the botId attribute from html and contact backend (B2) to confirm that this botId is valid, if so, A1 will connect to B1 to work with LLM.

For testing purposes, Leon has already provided a hard code `botId` and `sessionId` for us. My question is that is our FE ready to receive the attribute and handle it yet?

**Comment by vuson1709** [comment]
> 1. Have you prepared a list of what to do for the script tag to be hosted live? So whenever we have an update in the production's UI, it will be reflected on our client's website.

- In additional for this question, I've update small changes for the project, the previous version (1.0) didn't have the loading state. Then I make an update (version 1.1) with loading state. Then I upload the bundle file to a server (https://velvety-choux-22d460.netlify.app/widget/chat-widget.iife.js). Basically every time we upload a file to a server, it remains the same URL. 

So when customer inject the script with exact URL path above at the beginning, the product is automatically update when we update the product. They don't have to do anything else. 

`<script src="https://velvety-choux-22d460.netlify.app/widget/chat-widget.iife.js"></script>`

In version 1.0 (no loading state)
image


In version 1.1 (have loading state)
image


- If you want to know where is the server to store the bundle file come from. So I create a website and host on Netlify, that website acts like our marketing product. I takes advantage of Netlify because they allow us to add additional file while hosting website. So I also upload the bundle file also. That's why you can see the front page of website

> https://velvety-choux-22d460.netlify.app/
and this path contains our bundle file
https://velvety-choux-22d460.netlify.app/widget/chat-widget.iife.js

>

**Comment by vuson1709** [comment]
> For testing purposes, Leon has already provided a hard code `botId` and `sessionId` for us. My question is that is our FE ready to receive the attribute and handle it yet?

Good point, I will implement how can we handle `botID` and demo to you this evening before merging.

**Comment by vuson1709** [comment]
> For testing purposes, Leon has already provided a hard code `botId` and `sessionId` for us. My question is that is our FE ready to receive the attribute and handle it yet?

Hi @higherstates , on my newest [commit](https://github.com/AI-ChatBot-Yoga/ai-chatbot/pull/28/commits/75e7622ffcfed5bf276c7d86efc18cbfdc87e741), I added a code to read the `botId` attribute from the script. I injected the script to another [website](https://son-moviedatabase.netlify.app/) (assume that it's a client's website), it will read `botId` and store it in a variable with the same name, later it will be sent to server along with other data (message,...) to check if this id is valid. 
image
image

What do you think about this approach?

about `sessionId`, Leon told me that FE should generate a uuid. This uuid is live within a session. If user close website or click on clear chat button, a new uuid will be created. This is assigned to me on [this task](https://app.asana.com/0/1207458980633630/1207247627667191/f). Do you want me to implement this `sessionId` feature in this PR or on next PR?

File: src/components/ChatWindow/ChatWindow.tsx [hunk 1]
```diff
@@ -12,7 +12,11 @@ type Props = {
   onChatActivation: () => void
 }
 
-const BOT_ID = "71c0c33f-5952-43b1-8608-70bfe362f537" // Hard code for now, make it dynamic later
+const scriptTag = document.currentScript as HTMLScriptElement
+const botId = scriptTag?.getAttribute("botId")
+console.log("Script tag is:", scriptTag)
```
Comment (higherstates) [inline src/components/ChatWindow/ChatWindow.tsx:17]: please remove console logs in your next PR

**Review by higherstates — APPROVED** [review]
LGTM, please proceed in your next PR!

**Pull request merged**